Stream record test is flaky

**Home Assistant release with the issue:**
dev branch (0.91+)

**Last working Home Assistant release (if known):**


**Operating environment (Hass.io/Docker/Windows/etc.):**
<!--
Please provide details about your environment.
-->

**Component/platform:**
<!--
Please add the link to the documentation at https://www.home-assistant.io/components/ of the component/platform in question.
-->
stream

**Description of problem:**
`tests/components/stream/test_recorder.py::test_record_stream` test randomly failed CI environment

For example:
https://circleci.com/gh/home-assistant/home-assistant/1155

I am not understanding why @hunterjm want to test `segments == 3`, is `segments > 1` enough?

**Problem-relevant `configuration.yaml` entries and (fill out even if it seems unimportant):**
```yaml

```

**Traceback (if applicable):**
```
__________________________ test_record_stream[pyloop] __________________________

hass = <homeassistant.core.HomeAssistant object at 0x7fdb80c43be0>
hass_client = <function hass_client.<locals>.auth_client at 0x7fdb80c488c8>

    async def test_record_stream(hass, hass_client):
        """
        Test record stream.
    
        Purposefully not mocking anything here to test full
        integration with the stream component.
        """
        await async_setup_component(hass, 'stream', {
            'stream': {}
        })
    
        with patch(
                'homeassistant.components.stream.recorder.recorder_save_worker'):
            # Setup demo track
            source = generate_h264_video()
            stream = preload_stream(hass, source)
            recorder = stream.add_provider('recorder')
            stream.start()
    
            segments = 0
            while True:
                segment = await recorder.recv()
                if not segment:
                    break
                segments += 1
    
            stream.stop()
    
>           assert segments == 3
E           assert 2 == 3
E             -2
E             +3

tests/components/stream/test_recorder.py:44: AssertionError
---------------------------- Captured stderr setup -----------------------------
DEBUG:asyncio:Using selector: EpollSelector
------------------------------ Captured log setup ------------------------------
selector_events.py          54 DEBUG    Using selector: EpollSelector
----------------------------- Captured stderr call -----------------------------
INFO:homeassistant.loader:Loaded stream from homeassistant.components.stream
INFO:homeassistant.loader:Loaded http from homeassistant.components.http
INFO:homeassistant.setup:Setting up http
INFO:homeassistant.setup:Setup of domain http took 0.0 seconds.
DEBUG:homeassistant.core:Bus:Handling <Event component_loaded[L]: component=http>
INFO:homeassistant.setup:Setting up stream
DEBUG:homeassistant.core:Bus:Handling <Event service_registered[L]: domain=stream, service=record>
INFO:homeassistant.setup:Setup of domain stream took 0.0 seconds.
DEBUG:homeassistant.core:Bus:Handling <Event component_loaded[L]: component=stream>
INFO:homeassistant.components.stream:Started stream: <_io.BytesIO object at 0x7fdb80bf2150>
ERROR:homeassistant.components.stream.worker:Error demuxing stream: No dts in packet
INFO:homeassistant.components.stream:Stopped stream: <_io.BytesIO object at 0x7fdb80bf2150>
------------------------------ Captured log call -------------------------------
loader.py                  197 INFO     Loaded stream from homeassistant.components.stream
loader.py                  197 INFO     Loaded http from homeassistant.components.http
setup.py                   137 INFO     Setting up http
setup.py                   163 INFO     Setup of domain http took 0.0 seconds.
core.py                    541 DEBUG    Bus:Handling <Event component_loaded[L]: component=http>
setup.py                   137 INFO     Setting up stream
core.py                    541 DEBUG    Bus:Handling <Event service_registered[L]: domain=stream, service=record>
setup.py                   163 INFO     Setup of domain stream took 0.0 seconds.
core.py                    541 DEBUG    Bus:Handling <Event component_loaded[L]: component=stream>
__init__.py                175 INFO     Started stream: <_io.BytesIO object at 0x7fdb80bf2150>
worker.py                   81 ERROR    Error demuxing stream: No dts in packet
__init__.py                191 INFO     Stopped stream: <_io.BytesIO object at 0x7fdb80bf2150>
--------------------------- Captured stderr teardown ---------------------------
DEBUG:homeassistant.core:Bus:Handling <Event homeassistant_stop[L]>
INFO:homeassistant.components.stream:Stopped stream workers.
INFO:tests.common:Writing data to auth: {'version': 1, 'key': 'auth', 'data': {'users': [{'id': '4fe1a9e3972d46d6b1c509252f7fe38c', 'group_ids': ['system-admin'], 'is_owner': False, 'is_active': True, 'name': 'Mock User', 'system_generated': False}], 'groups': [{'id': 'system-admin', 'name': 'Administrators'}, {'id': 'system-users', 'name': 'Users'}, {'id': 'system-read-only', 'name': 'Read Only'}], 'credentials': [], 'refresh_tokens': [{'id': 'a2fd577e46c14446becde169c3ade7ea', 'user_id': '4fe1a9e3972d46d6b1c509252f7fe38c', 'client_id': 'https://example.com/app', 'client_name': None, 'client_icon': None, 'token_type': 'normal', 'created_at': '2019-03-31T06:28:29.490927+00:00', 'access_token_expiration': 1800.0, 'token': '51869528854d090511fd8de88807b8ed0c62249ab9558c850e74cc6fca4a9eae479551965378be0803bb73abe546625c1f619854f944c0dc611fdd3fe6e8a5b1', 'jwt_key': 'fb6e4e36b9c618fffad611d6c39f7f8c1c36761a5ad8dcde17512efd19a0e77ff55c865e29657f7821cea5a35435ea62d7fcc9fffeb6fb8fa6fc060960832483', 'last_used_at': '2019-03-31T06:28:29.491045+00:00', 'last_used_ip': None}]}}
DEBUG:homeassistant.core:Bus:Handling <Event homeassistant_close[L]>
---------------------------- Captured log teardown -----------------------------
core.py                    541 DEBUG    Bus:Handling <Event homeassistant_stop[L]>
__init__.py                108 INFO     Stopped stream workers.
common.py                  874 INFO     Writing data to auth: {'version': 1, 'key': 'auth', 'data': {'users': [{'id': '4fe1a9e3972d46d6b1c509252f7fe38c', 'group_ids': ['system-admin'], 'is_owner': False, 'is_active': True, 'name': 'Mock User', 'system_generated': False}], 'groups': [{'id': 'system-admin', 'name': 'Administrators'}, {'id': 'system-users', 'name': 'Users'}, {'id': 'system-read-only', 'name': 'Read Only'}], 'credentials': [], 'refresh_tokens': [{'id': 'a2fd577e46c14446becde169c3ade7ea', 'user_id': '4fe1a9e3972d46d6b1c509252f7fe38c', 'client_id': 'https://example.com/app', 'client_name': None, 'client_icon': None, 'token_type': 'normal', 'created_at': '2019-03-31T06:28:29.490927+00:00', 'access_token_expiration': 1800.0, 'token': '51869528854d090511fd8de88807b8ed0c62249ab9558c850e74cc6fca4a9eae479551965378be0803bb73abe546625c1f619854f944c0dc611fdd3fe6e8a5b1', 'jwt_key': 'fb6e4e36b9c618fffad611d6c39f7f8c1c36761a5ad8dcde17512efd19a0e77ff55c865e29657f7821cea5a35435ea62d7fcc9fffeb6fb8fa6fc060960832483', 'last_used_at': '2019-03-31T06:28:29.491045+00:00', 'last_used_ip': None}]}}
core.py                    541 DEBUG    Bus:Handling <Event homeassistant_close[L]>
```

**Additional information:**

--- a/tests/components/stream/test_recorder.py
+++ b/tests/components/stream/test_recorder.py
@@ -41,7 +41,7 @@ async def test_record_stream(hass, hass_client):
 
         stream.stop()
 
-        assert segments == 3
+        assert segments > 1
 
 
 async def test_recorder_timeout(hass, hass_client):

